File: dev-docs/snippets/databases/import-csv.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Import data from a CSV string into a database.
// JS-only rich pipeline — Swift offers only the pre-parsed `importRows` helper.
export async function importCsv(
  client: JsBaoClient,
  databaseId: string,
  csv: string,
) {
  // #region example
  const result = await client.databases.importCsv(databaseId, {
    csv,
    modelName: "Product",
    columnMap: { "Product Name": "name" },
    types: { price: "number" },
    onProgress: (progress) => console.log(progress.processed, "/", progress.total),
  });
  // #endregion example
  return result;
}

File: dev-docs/snippets/databases/import-csv.swift
import JsBaoClient

// Import data from a CSV string into a database, with column mapping, type
// coercion, and a progress callback — mirrors the JS rich CSV pipeline.
func importCsv(client: JsBaoClient, databaseId: String, csv: String) async throws {
  // #region example
  let result = try await client.databases.importCsv(
    databaseId: databaseId,
    options: CsvImportOptions(
      csv: csv,
      modelName: "Product",
      columnMap: ["Product Name": "name"],
      types: ["price": .number],
      onProgress: { progress in
        print(progress.processed, "/", progress.total)
      }
    )
  )
  // #endregion example
  _ = result
}
